### DISCUSSION TITLE
umi-max 在app.ts中export乾坤配置，执行的比getInitialStateAPI还更先，请问我想请求完接口动态改变乾坤的配置应该怎么做？

### DISCUSSION THREAD

**Discussion by kelisiWu123** [umijs/umi#12265] [Q&A]
export const   qiankun = (()=> {
  console.log('qian kun 配置加载')
  return {
    apps: [
      {
        name: 'App1',
        entry: `${someUrl}`,
      },
      {
        name: 'ptApp2',
        entry: `${getAppName()}`,
      },
    ],
  }
})()

**Comment by kelisiWu123** [comment]
思维局限了，原来把qiankun方法里面当成 getInitialState用就可以做到了

**Comment by bxter** [comment]
能截个图，学习下怎么用吗？

**Reply by fz6m** [reply]
楼上这里指的是使用 `export const qiankun = async () => {}` ，在里面调用 api ，进行 qiankun 运行时配置。
